tests/paz/pipelines/pix2pose_metric_test.py

```diff
@@ -0,0 +1,167 @@
+    cv2.setNumThreads(1)
+    cv2.setRNGSeed(777)
+    camera = Camera()
+    camera.intrinsics_from_HFOV(55, (480, 640, 3))
+    detector = PIX2POSEPowerDrill(camera, score_thresh=0.45, offsets=[0.25, 0.25], epsilon=0.015)
+    inferences = detector(image_with_powerdrill)
+    pose6D = inferences['poses6D'][0]
+    return pose6D
+
+
+@pytest.fixture
+def points3D():
+    X=np.linspace(1e-1,1e-3,  num=9128).reshape(9128,1)
+    Y=np.linspace(1e-2,1e-4,  num=9128).reshape(9128,1)
+    Z=np.linspace(1e-4,1e-6,  num=9128).reshape(9128,1)
+    points3D=np.hstack((X,Y,Z))
+    return points3D
+
+
+def test_quaternion_translation_shape(pix2pose_pose6d):
+    pred_quaternion, pred_translation = pix2pose_pose6d.quaternion, pix2pose_pose6d.translation
+    pred_pose = Pose6D(pred_quaternion, pred_translation)
+    pred_quaternion = get_quaternion(pred_pose)
+    assert (pred_quaternion.shape[0] == 4)
+    assert None not in pred_quaternion
+    pred_translation = get_translation(pred_pose)
+    assert (pred_translation.shape[0] == 3)
+    assert None not in pred_translation
+
+
+def test_transform_points3D(points3D, pose_A):
+    real_quaternion, real_translation = rotation_matrix_to_quaternion(
+        pose_A[0]), pose_A[1]
+    real_rotation = quaternion_to_rotation_matrix(real_quaternion)
+    assert (points3D.shape[1] == 3)
+    assert None not in points3D
+    transformed_points = transform_points3D(points3D, real_rotation, real_translation)
+    assert (transformed_points.shape[1] == 3)
+    assert None not in transformed_points
+
+
+def test_ADI_zero(pose_A, pose_B, pose_C, points3D):
+    a_quaternion, a_translation = rotation_matrix_to_quaternion(pose_A[0]), pose_A[1]
+    b_quaternion, b_translation = rotation_matrix_to_quaternion(pose_B[0]), pose_B[1]
+    c_quaternion, c_translation = rotation_matrix_to_quaternion(pose_C[0]), pose_C[1]     
+    b_pose = Pose6D(b_quaternion, b_translation)
+    a_pose = Pose6D(a_quaternion, a_translation)
+    c_pose = Pose6D(c_quaternion, c_translation)
+    adi_a = compute_ADI(a_pose, a_pose, points3D)
+    adi_b = compute_ADI(b_pose, b_pose, points3D)
+    add_c = compute_ADD(c_pose, c_pose, points3D)
+    assert adi_a==0
+    assert adi_b==0
+    assert add_c==0
+
+def test_ADD_zero(pose_A, pose_B, pose_C, points3D):
+    a_quaternion, a_translation = rotation_matrix_to_quaternion(pose_A[0]), pose_A[1]
+    b_quaternion, b_translation = rotation_matrix_to_quaternion(pose_B[0]), pose_B[1]
+    c_quaternion, c_translation = rotation_matrix_to_quaternion(pose_C[0]), pose_C[1]    
+    b_pose = Pose6D(b_quaternion, b_translation)
+    a_pose = Pose6D(a_quaternion, a_translation)
+    c_pose = Pose6D(c_quaternion, c_translation)
+    add_a = compute_ADD(a_pose, a_pose, points3D)
+    add_b = compute_ADD(b_pose, b_pose, points3D)
+    add_c = compute_ADD(c_pose, c_pose, points3D)
+    assert add_a==0
+    assert add_b==0
+    assert add_c==0
+
+        
+def test_ADI_ten(pose_A, pose_B, points3D):
```

Comment:
I think test_ADI_ten might not be a good test name. Maybe test_ADI_translation. I am actually unsure of how you were able to have a test for which the score is a specific value.

Reply:
resolved, (the pose B is just pose A , but 10 was added to an element of translation vector